ocaml/parsing/lexer.mll

```diff
@@ -384,6 +410,16 @@ let add_docstring_comment ds =
 
 let comments () = List.rev !comment_list
 
+let float ~maybe_hash lit modifier =
+  match maybe_hash with
+  | "#" -> HASH_FLOAT (lit, modifier)
+  | _ -> FLOAT (lit, modifier)
```

Comment:
This case presumably will always be the empty string? I think it's better to be explicit about that and then have a catch-all case that witnesses an error in the lexer.

Reply:
Sounds good to me.